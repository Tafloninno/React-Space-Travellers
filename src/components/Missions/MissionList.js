import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import AllMissions from './Missions';
import { getMissions } from '../../redux/missions/missionSlice';
import './Missions.css';

const MissionList = () => {
  const dispatch = useDispatch();
  const { missionList, isLoading } = useSelector((store) => store.mission);
  useEffect(() => {
    dispatch(getMissions());
  }, [dispatch]);

  if (isLoading) {
    return <div>Loading missions...Please Wait!</div>;
  }

  return (
    <div className="mission-list-container">
      <table>
        <thead>
          <tr>
            <th>Missions</th>
            <th>Description</th>
            <th>Status</th>
          </tr>
        </thead>
      </table>
      <table className="mission-list-table">
        <tbody>
          {missionList.map((mission) => (
            <AllMissions
              key={mission.mission_id}
              name={mission.mission_name}
              desc={mission.description}
              id={mission.mission_id}
            />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default MissionList;
