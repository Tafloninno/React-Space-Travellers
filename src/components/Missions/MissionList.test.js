import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import MissionList from './MissionList';

const mockStore = configureMockStore();

describe('MissionList', () => {
  it('renders correctly', () => {
    const store = mockStore({
      mission: {
        missionList: [
          {
            mission_id: '1',
            mission_name: 'Mission 1',
            description: 'Mission 1 Description',
          },
          {
            mission_id: '2',
            mission_name: 'Mission 2',
            description: 'Mission 2 Description',
          },
        ],
        isLoading: false,
      },
    });
    const tree = renderer
      .create(
        <Provider store={store}>
          <MissionList />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
