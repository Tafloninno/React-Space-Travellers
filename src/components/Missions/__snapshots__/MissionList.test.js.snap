// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionList renders correctly 1`] = `
<div
  className="mission-list-container"
>
  <table>
    <thead>
      <tr>
        <th>
          Missions
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
  </table>
  <table
    className="mission-list-table"
  >
    <tbody>
      <table>
        <tbody>
          <tr>
            <td
              className="name"
            >
              Mission 1
            </td>
            <td
              className="desc"
            >
              Mission 1 Description
            </td>
            <td
              className="btn-group"
            >
              <button
                className="not-member"
                disabled={true}
                type="button"
              >
                NOT A MEMBER
              </button>
              <button
                className="join-mission"
                onClick={[Function]}
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <tbody>
          <tr>
            <td
              className="name"
            >
              Mission 2
            </td>
            <td
              className="desc"
            >
              Mission 2 Description
            </td>
            <td
              className="btn-group"
            >
              <button
                className="not-member"
                disabled={true}
                type="button"
              >
                NOT A MEMBER
              </button>
              <button
                className="join-mission"
                onClick={[Function]}
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </tbody>
  </table>
</div>
`;
